### ISSUE TITLE
huobipro does not respond to watchOrderBook

### ISSUE THREAD

**Issue by davidmasin** [ccxt/ccxt#6646]
Hi, when I try to get orderbook using watchOrderBook for huobipro exchage, I get error. Could you check please?

- OS: Linux Unbuntu
- Programming Language version: javascript
- CCXT version: 6.13.4
- Exchange: huobipro
- Method: watchOrderBook

**Comment by kroitor** [comment]
Hi! 

> Could you check please?

We need more info from you, your version numbers, and the code snippet to reproduce it. As far as I can see it works in JS on my side in version 0.0.81.

**Comment by davidmasin** [comment]
Hi,

I use ccxt.pro 0.0.88
I tried to extract the code. It works for other exchanges, only with huobipro it does not.

Thanks 

const ccxtpro = require('ccxt.pro');
let exchangeSettings = {timeout: 10000};
let huobipro = new ccxtpro.huobipro(exchangeSettings);
huobipro.loadMarkets();
let limit = 10;
let marketSymbol = 'BTC/USDT';
let orderBook;
try {
  orderBook = await huobipro.watchOrderBook(marketSymbol, limit);
}
catch (err) {
  // I END-UP HERE!
}

**Comment by davidmasin** [comment]
Also, it works if watchOrderBook is replaced by basic ccxt fetchOrderBook

**Comment by kroitor** [comment]
Ok ) There's several errors in your code, see the comments below:

```JavaScript
const ccxtpro = require('ccxt.pro');

let exchangeSettings = {timeout: 10000}; // add enableRateLimit: true here

// you are required to enable the rate limiter, according to the Manual
let huobipro = new ccxtpro.huobipro(exchangeSettings); // https://github.com/ccxt/ccxt/wiki/Manual#rate-limit

huobipro.loadMarkets(); // this is an 'async' method that has to be 'await'ed!

let limit = 10;
let marketSymbol = 'BTC/USDT';
let orderBook;
try {
  orderBook = await huobipro.watchOrderBook(marketSymbol, limit);
}
catch (err) {
  // I END-UP HERE!
  // We need this output – it will really help resolve this issue quickly.
}
```

So, if you add the edits and corrections to that snippet, if you do this:

```JavaScript
'use strict';
const ccxt = require ('ccxt.pro');
(async () => {
    const exchangeSettings = {timeout: 10000, enableRateLimit: true}
    const exchange = new ccxtpro.huobipro(exchangeSettings)
    const limit = 10
    const symbol = 'BTC/USDT'
    await exchange.loadMarkets ()
    try {
        const orderbook = await exchange.watchOrderBook (symbol, limit)
    } catch (e) {
        console.log (e.constructor.name, e.message)
    }
}) ()
```

↑ Please, paste your output here, please. As far as I can see you are specifying the limit = 10, but to your disappointment, Huobi's API only accepts a `limit` of `undefined`, `20`, or `100`. At this time we can't really workaround that issue, because it's coming from the exchange itself, but we will look into meaningful workarounds to support custom `limit` values other than 20 or 100 for huobipro. If you set a different value – the library will raise an exception that tells you what to do. Everytime that happens you are supposed to read it and paste it here, because that is one of the most essential pieces of info (the actual output that you get), along with the code and versions, etc...

I suspect that you are getting that specific exception in your `catch`-clause. Need you to confirm this.

----

P.S. Use the GitHub's [markdown](https://guides.github.com/features/mastering-markdown/) to format your output for readability and padding, surround your code and/or output with triple backticks (&#096;&#096;&#096;), like shown below:

&#096;&#096;&#096;JavaScript

your code here

&#096;&#096;&#096;

---

&#096;&#096;&#096;

your output here

&#096;&#096;&#096;

**Comment by davidmasin** [comment]
Hi, thanks for quick reply.
It was the custom limit causing the issue.
But library has thrown an exception 

```
error TypeError: Class constructor  cannot be invoked without 'new' at huobipro.watchOrderBook (node_modules/ccxt.pro/js/huobipro.js:254:19)
```

It is really hard to understand what was wrong from this message. Maybe it could be more specific, such that this exchange only accepts specific values of limit, etc.

In any case, it is resolved for now from my side, thanks

**Comment by kroitor** [comment]
@davidmasin thank for reporting back, appreciate it! Indeed, there was a minor bug, that prevented a human-readable message and instead showed this:

```
error TypeError: Class constructor  cannot be invoked without 'new' at huobipro.watchOrderBook (node_modules/ccxt.pro/js/huobipro.js:254:19)
```

I've added a fix for that, so if you update to version 0.0.89 in a few minutes when the automatic build arrives and try a limit of 10 – it will at least throw a meaningful message from there.

Of course, the custom `limit` issue hasn't gone anywhere yet, since it's still on the exchange side, but we will try to resolve this as much as we can.

Thank you for using ccxt pro! Let us know if you have further issues or difficulties!
